backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxSparkPlanExecApi.scala

```diff
@@ -349,24 +349,30 @@ class VeloxSparkPlanExecApi extends SparkPlanExecApi {
     }
 
     def maybeAddAppendBatchesExec(plan: SparkPlan): SparkPlan = {
-      if (GlutenConfig.getConf.veloxCoalesceBatchesBeforeShuffle) {
-        VeloxAppendBatchesExec(plan, GlutenConfig.getConf.veloxMinBatchSizeForShuffle)
-      } else {
-        plan
+      plan match {
+        case shuffle: ColumnarShuffleExchangeExec
+            if !shuffle.useSortBasedShuffle &&
```

Comment:
Is this due to sort will do the coalesce behavior?

Reply:
No matter coalesced or not, sort will convert all inputs into rows, so it won't benefit from coalesce. The test result shows the coalesce op only brings extra overhead.